src/main/java/org/apache/sling/api/resource/ResourceResolver.java

```diff
@@ -842,4 +842,20 @@ Resource copy(final String srcAbsPath,
      */
     Resource move(final String srcAbsPath,
               final String destAbsPath) throws PersistenceException;
+
+    /**
+     * Returns a map to store temporary objects.
+     *
+     * This map is suited to store objects which share the same lifecycle as the
+     * resource resolver. The resource resolver itself does not use this map.
+     *
+     * The resource resolver will clear the map during {@link #close()}, so afterwards the map is empty.
+     * If a stored value implements the {@link #Closeable} interface, the ResourceResolver will invoke the
+     * <code>close()</code> of the value before clearing the map.
+     *
+     * @returns the property map
+     * @since 2.13.0
+     *
+     */
+    public @NotNull Map<String,Object> getPropertyMap();
```

Comment:
What about getSessionMap or getCacheMap? That way the intent is clearer from the name.

Reply:
I don't think that the use case is restricted to caching. The best name would probably getAttributesMap but we have read-only attributes already. For me, properties sounds good

Reply:
@kwin  Can we resolve this in one way or the other?

Reply:
Ok, let us stick with getPropertyMap()